Add shapeBorder option on App Bar

lsaudon opened the pull request:
I added borderRadius for a project.

Commit:
Add shapeBorder option on App Bar

Commit:
add shapeBorder option on App Bar

Commit:
Merge remote-tracking branch 'origin/master'

Hixie commented:
@lsaudon Thanks for your contribution! This will need a test to make sure we don't accidentally regress your change later, though.

cc @HansMuller @willlarche for review

willlarche commented:
Please add screenshots when you get a moment.

lsaudon commented:
`BorderRadius.only(
      bottomRight: Radius.circular(15.0),
      bottomLeft: Radius.circular(15.0),
    )`

Simple Appbar
<img width="424" alt="capture d ecran 2018-10-03 a 10 08 05" src="https://user-images.githubusercontent.com/25029876/46398157-a194bf00-c6f4-11e8-9c7c-40f281163fcd.png">

And with bottom
<img width="424" alt="capture d ecran 2018-10-03 a 10 09 52" src="https://user-images.githubusercontent.com/25029876/46398160-a3f71900-c6f4-11e8-83a9-70310e04893b.png">

HansMuller commented on packages/flutter/lib/src/material/app_bar.dart:
```diff
@@ -235,6 +236,15 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
 
+  /// The material's shape as well its shadow.
+  ///
+  /// If shape is non null, the [borderRadius] is ignored and the material's
```
There's no borderRadius here

lsaudon replied:
For the commentary I was inspired by the comment for the shape property in material.dart.

HansMuller commented on packages/flutter/lib/src/material/app_bar.dart:
```diff
@@ -235,6 +236,15 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
+  ///
+  /// A shadow is only displayed if the [elevation] is greater than
+  /// zero.
+  final ShapeBorder shapeBorder;
```
To be consistent with the button classes, Material, etc, this property should be called "shape"

lsaudon replied:
Yes,  this property should be called "shape".

HansMuller commented on packages/flutter/lib/src/material/app_bar.dart:
```diff
@@ -235,6 +236,15 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
   /// Defaults to 4, the appropriate elevation for app bars.
   final double elevation;
 
+  /// The material's shape as well its shadow.
```
Explain what the shape means in terms of the app bar

lsaudon replied:
It's the shapeborder. 
I got the wrong name.

HansMuller replied:
After you've fixed things up and added a test, use git push to update this pull-request.

Commit:
Rename shapeBorder to shape

Rename and change comment

willlarche commented:
So, I can't see from the screenshots but I assume the top corners are rounded too?

lsaudon commented:
It's a shapeborder. Yes, the top corners can be rounded too. https://docs.flutter.io/flutter/painting/ShapeBorder-class.html

willlarche commented:
Great! I don't think I'd recommend shaping the lower corners but we shouldn't forbid it. And the top ones are shaped in lots of apps these days.

willlarche commented:
So, no objections from Material product. Code up to @Hixie and @HansMuller to LGTM.

lsaudon commented:
I'm sorry, I'm sorry.
I can't do the tests. 
I have a lot of package errors with flutter and art in the project.

Commit:
Test AppBar with shape

lsaudon commented:
Is everything okay?

HansMuller commented on packages/flutter/test/material/app_bar_test.dart:
```diff
@@ -1422,4 +1422,26 @@ void main() {
+      return tester.widget<AppBar>(appBarFinder);
+    }
+
+    expect(getAppBarWidget().shape, roundedRectangleBorder);
```
Checking that the AppBar's shape parameter correctly initializes its shape property is useful, however you should also verify that the the AppBar Material widget's shape property is set.

You may find https://docs.flutter.io/flutter/flutter_test/CommonFinders/descendant.html to be useful for this.

HansMuller reviewed (changes requested):
Just a reminder: waiting on a small test change.

Commit:
Verify that the the AppBar Material widget's shape property is set


Revert "Verify that the the AppBar Material widget's shape property is set"

This reverts commit 372e67a707989641c2fa0f1f6b8913a9fcbb2afb.

Verify that the the AppBar Material widget's shape property is set

Commit:
Merge branch 'master' into master

HansMuller commented:
I will land this when the tests have finished.

HansMuller committed:
remove trailing whitespace

HansMuller committed:
More trailing white space
